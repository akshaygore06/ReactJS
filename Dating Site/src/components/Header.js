import React, { Component } from 'react';

export default class Header extends Component {

  render() {
    return (
      <div className="header">
          Lonely Hearts Dating Service
      </div>
    );
  }
}