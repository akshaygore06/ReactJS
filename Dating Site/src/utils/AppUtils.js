import _ from 'lodash';

const baseUrl = 'https://randomuser.me/api/';

export const HTTPStatus = {
    OK: 200,
    BAD_REQUEST: 400,
    UNAUTHORIZED: 401,
    NOT_FOUND: 404,
    INTERNAL_SERVER_ERROR: 500
};

const checkStatus = response => {
    if (response.status === 200) {
        return response;
    } else if (response.status === HTTPStatus.BAD_REQUEST) {
        const error = new Error("Server didn't like the request.");
        error.status = response.status;
        throw error;
    } else if (response.status === HTTPStatus.UNAUTHORIZED) {
        const error = new Error("Invalid email address or password.");
        error.status = response.status;
        throw error;
    } else if (response.status === HTTPStatus.NOT_FOUND) {
        const error = new Error("Server path doesn't exist.");
        error.status = response.status;
        throw error;
    } else {
        const message = response.statusText || 'HTTP error ' + response.status;
        const error = new Error(message);
        error.status = response.status;
        throw error;
    }
};

const fetchData = () => {
     return fetch(baseUrl + "?results=10")
    .then(checkStatus)
    .then(response => {
        return response.json();
    })
    .then(data => {
        return data.results;
    })
    .catch(error => console.log(error))
};

const filterData = (allData, ageRange, gender) => {
    const matchData = allData.filter( r => {
        const age = findAge(r);
        if( _.inRange(age, ageRange.min,ageRange.max+1)){
            if(gender === "any" || gender === r.gender) {
                return r;
            }
        }
        return null ;
    });
      return matchData;
}

const findAge = (r) =>{
    const birthdate = new Date(r.dob);
    const cur = new Date();
    const diff = cur-birthdate;
    const age = Math.floor(diff/31536000000);
    return age;
}

export default {
    fetchData,
    findAge,
    filterData,
};