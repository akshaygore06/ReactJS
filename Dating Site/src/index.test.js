// This unit test spec uses Jest: https://facebook.github.io/jest/docs/en/getting-started.html

import React from 'react';
import renderer from 'react-test-renderer';

import AppUtils from './utils/AppUtils';
import Header from './components/Header';

test('adds 1 + 2 to equal 3', () => {
    expect(1 + 2).toBe(3);
});

test('fetch data of 10 users', () => {
    expect.assertions(1);
    return AppUtils.fetchData().then(data => {
        expect(data.length).toBe(10);
    });
});

test('render a header', () => {
    const component = renderer.create(
        <Header />
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
});
