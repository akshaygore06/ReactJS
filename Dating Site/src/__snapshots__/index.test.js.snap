// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a header 1`] = `
<div
  className="header"
>
  Lonely Hearts Dating Service
</div>
`;

exports[`render a header1 1`] = `
<div
  className="header"
>
  Lonely Hearts Dating Service
</div>
`;
